### PR TITLE
Fix compiler warnings detected by gcc 13

### PR DESCRIPTION
**Pull request by cryptomilk**
This fixes several compiler warnings with gcc 13.

### CODE REVIEW

File: src/nvim/api/vim.c [hunk 1]
```diff
@@ -2241,7 +2241,7 @@ Dictionary nvim_eval_statusline(String str, Dict(eval_statusline) *opts, Error *
   if (highlights) {
     Array hl_values = ARRAY_DICT_INIT;
     const char *grpname;
-    char user_group[6];
+    char user_group[15];
```
Comment (lewis6991) [inline src/nvim/api/vim.c:2244]: Why was it 6 before, and why is it 15 now?

File: src/nvim/statusline.c [hunk 2]
```diff
@@ -1678,14 +1678,16 @@ int build_stl_str_hl(win_T *wp, char *out, size_t outlen, char *fmt, char *opt_n
...
-        stl_items[curitem].start = out_p + strlen(buf_tmp);
+        stl_items[curitem].start = out_p + buflen;
         curitem++;
         if (i == width) {
           str = buf_tmp;
           break;
         }
-        STRCAT(buf_tmp, p);
+        int rc = snprintf(buf_tmp + buflen, sizeof(buf_tmp) - buflen, "%s", p);
+        assert(rc > 0);
```
Comment (lewis6991) [inline src/nvim/statusline.c:1690]: Does this not also need `(void)rc;`?

File: src/nvim/api/vim.c (hunk 1 above)
Comment (cryptomilk) [reply src/nvim/api/vim.c:2244]: strlen(User) = 4
strlen(2147483647) = 10
10 + 4 + NIL

Comment (lewis6991) [reply src/nvim/api/vim.c:2244]: Can you add that as a comment.

Comment (cryptomilk) [reply src/nvim/api/vim.c]: I added a comment with that information.

File: src/nvim/statusline.c (hunk 2 above)
Comment (cryptomilk) [reply src/nvim/statusline.c:1691]: Added, thanks.

**Comment by cryptomilk** [comment]
This should be backported to 0.9.

File: src/nvim/api/vim.c (hunk 1 above)
Comment (seandewar) [reply src/nvim/api/vim.c]: [780ea84](https://github.com/neovim/neovim/pull/23239/commits/780ea845f9022b85a33b86f35c19b7d29594a466)

Refers to the `UserX` hl groups, so 4 chars for `User`, 1 for the digit, and 1 for the NUL presumably.

Comment (seandewar) [reply src/nvim/api/vim.c]: Naturally, GitHub didn't show me the new comments here :roll_eyes:

**Review by lewis6991 — APPROVED** [review]

File: src/nvim/api/vim.c [hunk 3]
```diff
@@ -2241,7 +2241,7 @@ Dictionary nvim_eval_statusline(String str, Dict(eval_statusline) *opts, Error *
   if (highlights) {
     Array hl_values = ARRAY_DICT_INIT;
     const char *grpname;
-    char user_group[6];
+    char user_group[15];  // strlen("User") + strlen("2147483647") + NIL
```
Comment (zeertzjq) [inline src/nvim/api/vim.c:2244]:
```suggestion
    char user_group[15];  // strlen("User") + strlen("2147483647") + NUL
```

Comment (cryptomilk) [reply src/nvim/api/vim.c]: Fixed. Thanks

File: src/nvim/statusline.c (hunk 2 above)
Comment (luukvbaal) [reply src/nvim/statusline.c:1691]: We can (and should IMO) use `xstrlcpy()` here rather than `snprintf()`.

File: src/nvim/ui_client.c [hunk 4]
```diff
@@ -65,7 +65,11 @@ uint64_t ui_client_start_server(int argc, char **argv)
 #ifdef MSWIN
     os_open_conin_fd();
 #else
-    dup(stderr_isatty ? STDERR_FILENO : STDOUT_FILENO);
+    int fd = dup(stderr_isatty ? STDERR_FILENO : STDOUT_FILENO);
+    if (fd < 0) {
+      return 0;
+    }
+    // FIXME: resource leak of fd
```
Comment (bfredl) [inline src/nvim/ui_client.c:72]: There is no "resource leak". this comment should be deleted.

**Pull request merged**